libraries/AP_Proximity/AP_Proximity.cpp

```diff
@@ -195,90 +200,46 @@ float AP_Proximity::distance_min() const
-    return drivers[primary_instance]->get_horizontal_distances(prx_dist_array);
-}
-
-// get number of layers.
-uint8_t AP_Proximity::get_num_layers() const
-{
-    if (!valid_instance(primary_instance)) {
-        return 0;
-    }
-    return drivers[primary_instance]->get_num_layers();
-}
-
-// get raw and filtered distances in 8 directions per layer. used for logging
-bool AP_Proximity::get_active_layer_distances(uint8_t layer, AP_Proximity::Proximity_Distance_Array &prx_dist_array, AP_Proximity::Proximity_Distance_Array &prx_filt_dist_array) const
-{
-    if (!valid_instance(primary_instance)) {
-        return false;
-    }
-    // get distances from backend
-    return drivers[primary_instance]->get_active_layer_distances(layer, prx_dist_array, prx_filt_dist_array);
+    Proximity_Distance_Array prx_filt_dist_array; // unused
+    return boundary.get_layer_distances(PROXIMITY_MIDDLE_LAYER, distance_max(), prx_dist_array, prx_filt_dist_array);
 }
 
 // get total number of obstacles, used in GPS based Simple Avoidance
 uint8_t AP_Proximity::get_obstacle_count() const
-{   
-    if (!valid_instance(primary_instance)) {
-        return 0;
-    }
-    return drivers[primary_instance]->get_obstacle_count();
+{
+    return boundary.get_obstacle_count();
 }
 
 // get vector to obstacle based on obstacle_num passed, used in GPS based Simple Avoidance
 bool AP_Proximity::get_obstacle(uint8_t obstacle_num, Vector3f& vec_to_obstacle) const
 {
-    if (!valid_instance(primary_instance)) {
-        return false;
-    }
-    return drivers[primary_instance]->get_obstacle(obstacle_num, vec_to_obstacle);
+    return boundary.get_obstacle(obstacle_num, vec_to_obstacle);
 }
 
 // returns shortest distance to "obstacle_num" obstacle, from a line segment formed between "seg_start" and "seg_end"
-// used in GPS based Simple Avoidance
+// returns FLT_MAX if it's an invalid instance.
 bool AP_Proximity::closest_point_from_segment_to_obstacle(uint8_t obstacle_num, const Vector3f& seg_start, const Vector3f& seg_end, Vector3f& closest_point) const
 {
-    if (!valid_instance(primary_instance)) {
-        return false;
-    }
-    return drivers[primary_instance]->closest_point_from_segment_to_obstacle(obstacle_num, seg_start, seg_end, closest_point);
+    return boundary.closest_point_from_segment_to_obstacle(obstacle_num , seg_start, seg_end, closest_point);
 }
 
 // get distance and angle to closest object (used for pre-arm check)
 //   returns true on success, false if no valid readings
 bool AP_Proximity::get_closest_object(float& angle_deg, float &distance) const
 {
-    if (!valid_instance(primary_instance)) {
-        return false;
-    }
-    // get closest object from backend
-    return drivers[primary_instance]->get_closest_object(angle_deg, distance);
+    return boundary.get_closest_object(angle_deg, distance);
 }
 
-// get number of objects, used for non-GPS avoidance
+// get number of objects, angle and distance - used for non-GPS avoidance
```

Comment:
this looks like a copy-paste error in the comment.  the get_object_count() method doesn't return angle and distance.